### PR TITLE
Splitdate

### PR DESCRIPTION
**Pull request by shadrackkibet**
@Patowhiz this is ready for you to merge. I have now added `s_quarter` parameter and `fiscal_start` on the `split_date` method. I also added `with_year` as a parameter but that could be implemented later.

### CODE REVIEW

**Comment by Patowhiz** [comment]
This pull request has calculator file changes

**Comment by shadrackkibet** [comment]
I think your branch is behind from the master branch changes. If you merge this it won't have any issue.

File: instat/static/InstatObject/R/instat_object_R6.R [hunk 1]
```diff
@@ -1150,8 +1150,8 @@ instat_object$set("public","create_factor_data_frame", function(data_name, facto
...
-  self$get_data_objects(data_name)$split_date(col_name = col_name , week = week, month_val = month_val,  month_abbr = month_abbr, month_name = month_name, weekday_val = weekday_val, weekday_abbr = weekday_abbr,  weekday_name =  weekday_name, day = day, year = year, day_in_month = day_in_month, day_in_year = day_in_year,  leap_year =  leap_year, day_in_year_366 = day_in_year_366, dekade = dekade, pentad = pentad, quarter = quarter, s_doy = s_doy, s_year = s_year, s_start_day_in_month = s_start_day_in_month, s_start_month = s_start_month)
+instat_object$set("public","split_date", function(data_name, col_name = "", year = FALSE, day = FALSE, week = FALSE,  month_val = FALSE, month_abbr = FALSE, month_name = FALSE, weekday_val = FALSE, weekday_abbr = FALSE, weekday_name = FALSE, day_in_month = FALSE, day_in_year = FALSE,  leap_year = FALSE, day_in_year_366 = FALSE, dekade = FALSE, pentad = FALSE, quarter = FALSE, s_quarter = FALSE, with_year = FALSE,  s_doy = FALSE, s_year = FALSE, s_start_day_in_month = 1, s_start_month = 8) {
```
Comment (Patowhiz) [inline instat/static/InstatObject/R/instat_object_R6.R]: @shadrackkibet I can see you added parameter ` with_year = FALSE` here. What is it for?

**Comment by Patowhiz** [comment]
In the dialog, you have added `ucrNudFiscalStart `control. Was this Danny's suggestion? Cause I think it's purpose was to be done by `ucrInputComboBoxMonth `control

File: instat/static/InstatObject/R/instat_object_R6.R (hunk 1 above)
Comment (shadrackkibet) [reply instat/static/InstatObject/R/instat_object_R6.R]: That is for adding a year together with the quarter separated by a dot.  We discussed with Danny and that will be added in future.

**Comment by shadrackkibet** [comment]
> In the dialog, you have added ucrNudFiscalStartcontrol. Was this Danny's suggestion? Cause I think it's purpose was to be done by ucrInputComboBoxMonthcontrol

You are right actually that should  be removed.

**Pull request merged**